modules/yarpc/yarpc.go

```diff
@@ -281,16 +247,17 @@ func newYARPCModule(
 		statsClient: newStatsClient(host.Metrics()),
 		log:         ulog.Logger(context.Background()).With(zap.String("module", host.Name())),
 	}
-	if err := host.Config().Scope("modules").Get(host.Name()).PopulateStruct(&module.config); err != nil {
-		return nil, errs.Wrap(err, "can't read inbounds")
-	}
+
+	val := host.Config().Scope("modules").Get(host.Name()).Value()
 
 	// iterate over inbounds
-	transportsIn, err := prepareInbounds(module.config.Inbounds, host.Name())
+	c, err := cfg.New().LoadConfig(val)
```

Comment:
Yes, in case of :
```yaml
modules:
  hello:
   inbounds:
     http:
       address: ":8080"
     redis:
       queueKey: requests
       processingKey: processing
   transports:
     redis:
       address: localhost:6379
```
We'll pass this map(which is map[string]map[string]map[string]string :) ):
```yaml
   inbounds:
     http:
       address: ":8080"
     redis:
       queueKey: requests
       processingKey: processing
   transports:
     redis:
       address: localhost:6379
```